### PR TITLE
Improve LFS tests + fix lfs url refs + keep path upper/lowercase in db.

### LINKED ISSUE #3101: LFS: unable to push objects tracked by git LFS
- Gitea version (or commit ref): master (c7fb6e30870ea1abff13a8214107e0747d293320)
- Git version: 2.11.0
- Operating system: Ubuntu 17.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant  (LFS seems to not be available)
- Log gist: https://gist.github.com/flozz/e235a55a88d4f6d74eed66fe8c9d7d47

## Description

I am not able to push objects tracked by git LFS anymore (tester on latest master). Steps to reproduce:

```
$ git git clone http://localhost:3000/test/test-lfs.git
Cloning into 'test-lfs'...
warning: You appear to have cloned an empty repository.

$ cd test-lfs

$ git lfs install
Updated git hooks.
Git LFS initialized.

$ git lfs track "*.txt" --lockable
Tracking "*.txt"

$ echo test > foobar.txt

$ git add -A
warning: CRLF will be replaced by LF in .gitattributes.
The file will have its original line endings in your working directory.

$ git commit -m test
[master (root-commit) 8fdb137] test
 2 files changed, 4 insertions(+)
 create mode 100644 .gitattributes
 create mode 100644 foobar.txt

$ git push
Username for 'http://localhost:3000': test
Password for 'http://test@localhost:3000': 
Username for 'http://localhost:3000': test
Password for 'http://test@localhost:3000': 
Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.http://localhost:3000/test/test-lfs.git/info/lfs.locksverify true
Git LFS: (0 of 1 files) 0 B / 5 B                                                                               
LFS: Repository or object not found: http://localhost:3000/test/test-lfs/info/lfs/objects/f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
Check that it exists and that you have proper access to it
error: failed to push some refs to 'http://localhost:3000/test/test-lfs.git'
```

### PR DESCRIPTION
**Pull request by sapk**
All in the title. 

Fix #3101

### CODE REVIEW

**Comment by sapk** [comment]
Need https://github.com/dockhippie/golang/pull/2

**Comment by sapk** [comment]
I need to setup a separate repo to not influence other repos tests since clone after will not work because the server is not really started and can't get file for lfs.

**Comment by strk** [comment]
LGTM

**Comment by lunny** [comment]
LGTM

**Comment by lunny** [comment]
CI failed.

File: integrations/git_test.go [hunk 1]
```diff
@@ -37,24 +42,144 @@ func onGiteaWebRun(t *testing.T, callback func(*testing.T, string)) {
...
+					Private:     false,
+					Gitignores:  "",
+					License:     "WTFPL",
+					Readme:      "Default",
+				})
+				session.MakeRequest(t, req, http.StatusCreated)
+			})
+
+			u.Path = "user2/repo-tmp-17.git"
+			u.User = url.UserPassword("user2", "password")
```
Comment (ethantkoenig) [inline integrations/git_test.go]: nit: use the `userPassword` constant (also in 162).

**Review by ethantkoenig — COMMENTED** [review]
One general comment is that there is a lot of repeated boilerplate (e.g. making commits, pushing). While it would be nice to consolidate, I'm guessing there are other tests with similar boilerplate, so it's probably best to do wide-scale refactoring in a separate PR.

**Comment by lunny** [comment]
@sapk maybe you could force push once since the Drone just recovered.

**Comment by lafriks** [comment]
@sapk test failed

**Comment by lunny** [comment]
@sapk CI still failed because of fmt-check

**Comment by sapk** [comment]
@lunny I know but I fix it but drone keep to stay on that commit. I retry to start multiple time, squash the commit fixing fmt ... with no luck

**Comment by sapk** [comment]
@lafriks thx, but drone seems broken and block on an unrelated test since 1hour ...

**Comment by lafriks** [comment]
@sapk ~~please submit backport to 1.3 branch~~ this bug was just in master and not in 1.3.0, right?

**Comment by sapk** [comment]
@lafriks It should not be in 1.3

**Comment by sapk** [comment]
thx all for the help.

**Pull request merged**